datalad/support/gitrepo.py

```diff
@@ -1577,18 +1583,18 @@ def get_hexsha(self, commitish=None, short=False):
             return stdout[0]
 
     @normalize_paths(match_return_type=False)
-    def get_last_commit_hash(self, files):
+    def get_last_commit_hexsha(self, files):
         """Return the hash of the last commit the modified any of the given
         paths"""
         try:
-            stdout, stderr = self._git_custom_command(
-                files,
-                ['git', 'log', '-n', '1', '--pretty=format:%H'],
-                expect_fail=True)
-            commit = stdout.strip()
-            return commit
+            commit = self.call_git(
```

Comment:
Given that this is a call within GitRepo, I think we might as well use `self._git_custom_command` to avoid an additional function call and a couple of additional attribute lookups.

Also, should this be called with `expect_fail=True` since it has the expected unborn branch failure?

Reply:
`expect_fail` is fixed.

Re not using `call_git()`: I'd prefer to move away from the old internal helper. My goal is to see what functionality we actually need, and ultimately have runners that are optimized for what they need to do. If we keep call the all-in-one method, the cognitive complexity of the problem stays where it is -- which is too high to do anything about it. Otherwise someone would have started with it already IMHO.

Reply:
OK.  I don't know how likely it is that using `call_git` instead of `_git_custom_command` really helps in that regard, but it doesn't cost much.